lib/ui/window/platform_message_response_dart.cc

```diff
@@ -19,30 +19,40 @@ namespace {
 // Avoid copying the contents of messages beyond a certain size.
 const int kMessageCopyThreshold = 1000;
 
+class DataWrapper {
+ public:
+  DataWrapper(void* data) { data_ = data; }
+
+  ~DataWrapper() { free(data_); }
+
+ private:
+  void* data_;
+};
+
 void MessageDataFinalizer(void* isolate_callback_data,
                           Dart_WeakPersistentHandle handle,
                           void* peer) {
-  std::vector<uint8_t>* data = reinterpret_cast<std::vector<uint8_t>*>(peer);
+  DataWrapper* data = reinterpret_cast<DataWrapper*>(peer);
   delete data;
 }
 
-Dart_Handle WrapByteData(std::vector<uint8_t> data) {
-  if (data.size() < kMessageCopyThreshold) {
+Dart_Handle WrapByteData(std::unique_ptr<fml::Mapping> mapping) {
+  size_t size = mapping->GetSize();
+  if (size < kMessageCopyThreshold) {
```

Comment:
This `if` branch is unnecessary as we're always calling `memcpy`: whether inside `ToByteData`, or inside the `else` branch, or inside the old `WrapByteData(std::unique_ptr<fml::Mapping> mapping)` function.